### DISCUSSION TITLE
Run app in browser, mocking electron processes

### DISCUSSION THREAD

**Discussion by kkerti** [alex8088/electron-vite#391] [General]
Hi, 
we have an app based on electron-vite. Running great in electron. 
Wanted to ask, what would be the best way to mock the electron parts to allow the app to run in the browser?
My goal would be to mock some of the `main` and `preload` functions in separate files for example in `web-main.ts` and `web-preload.ts`. It would be great if I could create a web build and electron build separetely. 
Any tip would be appreciated, how such a setup could be setup.
Here is the repo: https://github.com/intechstudio/grid-editor

**Comment by alex8088** [comment]
You can add a separate vite configuration file for the renderer and use vite to build. In addition, you need to set root to `src/renderer` in vite config
